File: BOJ/BOJ_13904_과제.py
N = int(input())
work = [list(map(int, input().split())) for _ in range(N)]
work.sort(reverse=True)
MAX = work[0][0]
answer = 0

for i in range(MAX, 0, -1):
    tmp = []
    for j in range(len(work)):
        if i <= work[j][0]:
            tmp.append([work[j][1], j])
    if len(tmp) > 0:
        tmp.sort(reverse=True)
        answer += tmp[0][0]
        del work[tmp[0][1]]

print(answer)

File: BOJ/BOJ_13904_과제.cpp
#include <bits/stdc++.h>
using namespace std;
typedef pair<int, int> pii;

int N, MAX, answer;
vector<pii> work;

void init() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

int main(int argc, char const *argv[]) {
    init();
    cin >> N;
    for (int i = 0; i < N; i++) {
        int left, score;
        cin >> left >> score;
        work.push_back({left, score});
    }
    sort(work.rbegin(), work.rend());
    MAX = work[0].first;

    for (int idx = MAX; idx >= 1; idx--) {
        vector<pii> tmp;
        for (int i = 0; i < work.size(); i++) {
            if (idx <= work[i].first) {
                tmp.push_back({work[i].second, i});
            }
        }
        if (tmp.size() != 0) {
            sort(tmp.begin(), tmp.end());
            answer += tmp.back().first;
            work.erase(work.begin() + tmp.back().second);
        }
    }
    cout << answer << '\n';
    return 0;
}